src/CMakeLists.txt

```diff
@@ -1,6 +1,12 @@
 configure_file ( version.h.in version.h ESCAPE_QUOTES @ONLY )
 
-target_sources ( ${PROJECT_NAME} PRIVATE main.cpp server.cpp packet.cpp attribute.cpp vendor_attribute.cpp utils.cpp dictionaries.cpp)
+add_library ( radproto STATIC )
+
+target_sources ( radproto PRIVATE server.cpp packet.cpp attribute.cpp vendor_attribute.cpp utils.cpp dictionaries.cpp)
+
+target_link_libraries ( ${PROJECT_NAME} radproto )
+
+target_include_directories ( radproto PUBLIC ${CMAKE_CURRENT_SOURCE_DIR} )
```

Comment:
This statement is not needed. By default, you can include headers from the same directory without explicitly stating that to a compiler. This statement would be useful if you use IMPORTED targets. For now you only need it for the sample server target.